File: app/src/main/java/com/bigscreen/mangindo/custom/TopCropImageView.java
package com.bigscreen.mangindo.custom;


import android.content.Context;
import android.graphics.Matrix;
import android.support.v7.widget.AppCompatImageView;
import android.util.AttributeSet;

public class TopCropImageView extends AppCompatImageView {

    public TopCropImageView(Context context) {
        super(context);
        setScaleType(ScaleType.MATRIX);
    }

    public TopCropImageView(Context context, AttributeSet attrs) {
        super(context, attrs);
        setScaleType(ScaleType.MATRIX);
    }

    public TopCropImageView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        setScaleType(ScaleType.MATRIX);
    }

    @Override
    protected boolean setFrame(int l, int t, int r, int b) {
        if (getDrawable() != null) {
            Matrix matrix = getImageMatrix();
            float scaleFactor = getWidth() / (float) getDrawable().getIntrinsicWidth();
            matrix.setScale(scaleFactor, scaleFactor, 0, 0);
            setImageMatrix(matrix);
        }
        return super.setFrame(l, t, r, b);
    }
}

File: app/src/main/java/com/bigscreen/mangindo/common/custom/TopCropImageView.kt
package com.bigscreen.mangindo.common.custom

import android.content.Context
import android.util.AttributeSet
import androidx.appcompat.widget.AppCompatImageView

class TopCropImageView @JvmOverloads constructor(
        context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0
) : AppCompatImageView(context, attrs, defStyleAttr) {

    init {
        scaleType = ScaleType.MATRIX
    }

    override fun setFrame(l: Int, t: Int, r: Int, b: Int): Boolean {
        if (drawable != null) {
            val matrix = imageMatrix
            val scaleFactor = width / drawable.intrinsicWidth.toFloat()
            matrix.setScale(scaleFactor, scaleFactor, 0f, 0f)
            imageMatrix = matrix
        }
        return super.setFrame(l, t, r, b)
    }
}
